service/producer/service.go

```diff
@@ -15,29 +19,40 @@ type Service struct {
 	ctx    context.Context
 	srv    *server.Server
 	health *health.Core
+	config config.Config
 }
 
 func NewService(ctx context.Context) *Service {
 	return &Service{ctx: ctx}
 }
 
 func (svc *Service) Start() {
-	// Define server handlers
 
 	healthCore, err := health.NewCore(nil)
 	if err != nil {
 		panic(err)
 	}
 
+	producer := NewProducer(boot.Config.Producer)
```

Comment:
Make all singleton initializations in boot and pass it here. Essentially follow dependency injection pattern

Reply:
Yup this is a singleton initialization.